components/eam/cime_config/testdefs/testmods_dirs/eam/condidiag_dcape/user_nl_eam

```diff
@@ -3,7 +3,7 @@
  qoi_chkpt = 'RAD', 'PACEND','DYNEND','DEEPCU', 'STCLD',
 
  qoi_name = 'CAPE','dCAPE'
- qoi_nver =  1,     72,
+ qoi_nver =  1,     80,
```

Comment:
To make this more robust we should change this so that the the levels default to something like -1, and then internally make sure they get set to 1 and pver. If that's an acceptable alternative I'm happy to implement it.

Reply:
Agreed setting to something like -1 and inside the code to resolve it to pver would be more elegant. That said, for users who do want to use this condidiag tool, it is probably more straightforward for them to specify the actual # of model levels they run with.

Reply:
right, but if most of the time they just want the upper and lower limit of the vertical levels then having it set internally makes sense, and the edge case where a user wants something else can be handled by editing these.

Reply:
Hi @huiwanpnnl , the condidiag related tests failed at runtime for 80-level configurations.  Reseting qoi_nver to 80 from 72 will allow the tests to run. Should qoi_nver value always be the same as the vertical dimension size of the corresponding diagnostic variable?

Reply:
Thank you, @huiwanpnnl . Your suggested changes are straightforward. I am going to put them in. BTW, would it be better to keep the initialization/default value (0) unchanged at line 219 and 228, as the new implementation would interpret `-1` as `pver`. That is just to avoid confusion as the where statement has strict upper bound in the array scope.

Reply:
@wlin7, your suggestion sounds good to me. In that scenario, `-1` will be interpreted as an intentional alias for `pver`, while 0 will indicate an unset value.

Reply:
Exactly, @huiwanpnnl, thanks.. index `-1`  as used in  accessing the last element of an array in some languages.  This is mainly for pre-defined testing cases. Users can still specify the actual number of levels in practical use, which would be equivalent. The changes are in [ee14d67](https://github.com/E3SM-Project/E3SM/pull/6054/commits/ee14d677a8c4c3155c86000fa14e1bb7f87d7560) and verified working well.